### PR TITLE
CHIP-SPV Integration

### PR DESCRIPTION
**Pull request by pvelesko**
@gargrahul

### CODE REVIEW

**Comment by pvelesko** [comment]
@gargrahul

**Comment by gargrahul** [comment]
@pvelesko  Could you please rebase your branch with the latest develop?

**Comment by pvelesko** [comment]
* Noticed the use of `#pragma once` in `hipRTC.h` so decided to switch to using it instead of ifdef include guards. This is simply due to a large number of iddefs already in the code. 
* more consistent use of includes <..>/"..."

File: include/hip/hip_common.h [hunk 1]
```diff
@@ -54,9 +53,13 @@ THE SOFTWARE.
...
-// Auto enable __HIP_DEVICE_COMPILE__ if compiled in HCC or NVCC device path
-#if (defined(__HCC_ACCELERATOR__) && __HCC_ACCELERATOR__ != 0) ||                                  \
-    (defined(__CUDA_ARCH__) && __CUDA_ARCH__ != 0)
+// // Auto enable __HIP_DEVICE_COMPILE__ if compiled in HCC or NVCC device path
+// #if (defined(__HCC_ACCELERATOR__) && __HCC_ACCELERATOR__ != 0) || \
+//     (defined(__CUDA_ARCH__) && __CUDA_ARCH__ != 0)
+// #define __HIP_DEVICE_COMPILE__ 1
+// #endif
+// TODO
+#if (defined(__clang__) && defined(__HIP_DEVICE_COMPILE__))
```
Comment (pvelesko) [inline include/hip/hip_common.h:62]: This entire section I'm confused about @gargrahul

File: include/hip/hip_runtime_api.h [hunk 2]
```diff
@@ -6965,10 +6989,11 @@ static inline hipError_t hipMallocManaged(T** devPtr, size_t size,
     return hipMallocManaged((void**)devPtr, size, flags);
 }
 
-#endif
-#endif
+#endif // __cplusplus
 
+#if (defined(__HIP_PLATFORM_HCC__) || defined(__HIP_PLATFORM_AMD__)) && !(defined(__HIP_PLATFORM_NVCC__) || defined(__HIP_PLATFORM_NVIDIA__))
 #include <hip/amd_detail/amd_hip_runtime_pt_api.h>
+#endif // TODO
```
Comment (pvelesko) [inline include/hip/hip_runtime_api.h]: in CHIP-SPV we integrated this into `hip_spirv_runtime.h` why have this separate?

**Pull request closed without merging**